modules/post/windows/gather/credentials/windows_autologin.rb

```diff
@@ -58,24 +56,16 @@ def run
     du2 = registry_getvaldata(logon_key, "AltDefaultUserName")   || ''
     dp2 = registry_getvaldata(logon_key, "AltDefaultPassword")   || ''
 
-    if do1 != '' and  du1 != '' and dp1 == '' and al == '1'
+    if do1 != '' && du1 != '' && dp1 == ''
```

Comment:
In testing against the free Windows 7 IE test VM Microsoft offers, this specific conditional worked OK (where the dp1 value shows up as empty even though the system does have a password set, and the module goes on to save the discovered creds), but testing against 2008 R2 (where dp1's value does retrieve the actual password) fails this conditional since dp1 == '' isn't true, leading the creds to not be saved and the "The Host #{host_name} is not configured to have AutoLogon password" output to appear.  We could remove the dp1 check altogether or make it something like the following to preserve the existing behavior:

`if do1 != '' &&  du1 != '' && (dp1 != '' || (dp1 == '' && al == '1'))`

And make the same sort of update to the conditional on line 65, of course.

What do you think, @FireFart?  And thanks for the cleanup!  :+1: